themes/developer/developer.theme.sh

```diff
@@ -0,0 +1,229 @@
+		printf "OPI5P"
+	else
+		printf "linux"
+	fi
+}
+
+function cpu_load {
+	# Ejecutar el comando top en modo batch, filtrar por el nombre de usuario actual y almacenar la salida en la variable 'output'
+	local output=$(top -b -n 1 -u $USER | grep "Cpu(s)" &)
+
+	# Extraer el porcentaje de carga de la CPU excluyendo el estado "idle" usando awk
+	local cpu_load=$(echo "$output" | awk '{print 100.0-$8}' | cut -d '.' -f 1 &)
+
+	# Imprimir la carga de la CPU
+	printf "${cpu_load}"
+}
+
+function getCpuLoad {
+	local current_cpu_load=$(cpu_load &)
+
+	local color="${_omb_prompt_reset_color}"
+	# Condicional para verificar los rangos
+	if ((current_cpu_load <= 40)); then
+		color="${_omb_prompt_teal}"
+	elif ((current_cpu_load >= 41 && current_cpu_load <= 50)); then
+		color="${_omb_prompt_reset_color}"
+	elif ((current_cpu_load >= 51 && current_cpu_load <= 60)); then
+		color="${_omb_prompt_olive}"
+	elif ((current_cpu_load >= 61 && current_cpu_load <= 75)); then
+		color="${_omb_prompt_red}"
+	elif ((current_cpu_load >= 76)); then
+		color="${_omb_prompt_red}!"
+	fi
+	return_delimited $1 "cpuload" "${color}${current_cpu_load}"
+}
+
+function getCpuTemp {
+	local temp_in_c
+	local currentPlatform=$(currentPlatform)
+
+	if ((currentPlatform == "linux")); then
+		temp_in_c=$(genericLinuxTemp)
+	elif ((currentPlatform == "OPI5P")); then
+		temp_in_c=$(OPi5p_Temp)
+	fi
+
+	local color
+	# Condicional para verificar los rangos
+	if ((temp_in_c >= 1 && temp_in_c <= 40)); then
+		color="${_omb_prompt_teal}"
+	elif ((temp_in_c >= 41 && temp_in_c <= 50)); then
+		color="${_omb_prompt_reset_color}"
+	elif ((temp_in_c >= 51 && temp_in_c <= 60)); then
+		color="${_omb_prompt_olive}"
+	elif ((temp_in_c >= 61 && temp_in_c <= 75)); then
+		color="${_omb_prompt_red}"
+	elif ((temp_in_c >= 76 && temp_in_c)); then
+		color="${_omb_prompt_red}!"
+	fi
+	return_delimited $1 "cputemp" "${color}${temp_in_c}°"
+}
+
+# this should work on every "new" linux distro
+function getDefaultIp {
+	# Obtiene el nombre de la interfaz de red activa
+	local interface=$(ip -o -4 route show to default | awk '{print $5}')
+
+	# Obtiene la dirección IP de la interfaz de red activa
+	local ip_address=$(ip -o -4 address show dev "$interface" | awk '{split($4, a, "/"); print a[1]}')
+
+	return_delimited $1 "ip" "${ip_address}"
+}
+
+# prompt constructor
+function _omb_theme_PROMPT_COMMAND() {
+	# start_time=$(($(date +%s%N) / 1000000))
+
+	#cputemp=$(getCpuTemp &)
+
+	#cpuload=$(getCpuLoad &)
+
+	#pyversion=$(py_version &)
+	#nodeversion=$(node_version &)
+	#goversion=$(go_version &)
+
+	#defaultip=$(getDefaultIp &)
+
+	#wait
+	# NEW way using paralellism
+	# this throws all inside the $() as a new thread but awaits the execution end_time
+	# so this takes the same time as the slower function
+	values=$(
+		getCpuLoad true & # this is very slow
+		getCpuTemp true &
+		py_version true &
+		node_version true &
+		go_version true &
+		getDefaultIp true &
```

Comment:
Is this robust? What happens if the outputs from multiple processes are interleaved? When the system is busy or the user is in the system where the atomicity of the IO operations is not ensured, the output can be mixed.

Reply:
4c65d8dd262589e263541bd8ccf9a329e7be9427